src/Crate/PDO/PDOStatementImplementationPhp8.php

```diff
@@ -35,9 +35,9 @@ trait PDOStatementImplementationPhp8
      * @param int   $mode
      * @param mixed ...$args
      *
-     * @return bool
+     * @return true
      */
-    public function setFetchMode(int $mode, ...$args)
+    public function setFetchMode(int $mode, ...$args): true
```

Comment:
```
PHP Fatal error:  Cannot use 'Crate\PDO\true' as class name as it is reserved in /home/runner/work/crate-pdo/crate-pdo/src/Crate/PDO/PDOStatementImplementationPhp8.php on line 40
```
-- https://github.com/crate/crate-pdo/actions/runs/13292872172/job/37118008326?pr=164#step:7:11

Reply:
if i set it to bool i get the error:

```
Return type of Crate\PDO\PDOStatementImplementation::setFetchMode(int $mode, ...$args): bool should either be compatible with PDOStatement::setFetchMode(int $mode, mixed ...$args): true, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice // vendor/crate/crate-pdo/src/Crate/PDO/PDOStatement.php:35
```

besides that, my phpstorm is throwing errors because of this mix of php7 and php8 files:

![image](https://github.com/user-attachments/assets/02bfcb91-f667-4ddf-a1bf-0ab3a979226a)

how does this package know which files should be used: 7 or 8 ? 🤔

Reply:
the deprecations notice I get with my `php 8.4.3` is strange and confuses me. I guess "true" is okay as a return value in PHP 8.4 while in 8.1 it´s not. 🤔 to be honest: I didn´t think much about it because I just wanted this package to work so I changed it to "true", no warning, fine 😏

Reply:
Thanks for elaborating. Does it also work without changing anything?

Reply:
nope ➡️ https://github.com/crate/crate-pdo/issues/160#issuecomment-2654739605

Reply:
btw.. if you like to test this yourself, here´s the docker image I am using: `michabbb/frankenphplaravel:8.4`